### DISCUSSION TITLE
have error in composer

### DISCUSSION THREAD

**Discussion by vahidalvandi** [spatie/laravel-honeypot#78] [Q&A]
```
Problem 1
    - spatie/laravel-honeypot[3.0.0, ..., 3.0.1] require illuminate/contracts ^8.0 -> found illuminate/contracts[v8.0.0, ..., 8.x-dev] but these were not loaded, likely because it conflicts with another require.
    - Root composer.json requires spatie/laravel-honeypot ^3.0 -> satisfiable by spatie/laravel-honeypot[3.0.0, 3.0.1].
```

**Comment by stebogit** [comment]
I have the same issue. Using Laravel 6.x and can't install but **v2.0**.

Looks like [this commit](https://github.com/spatie/laravel-honeypot/commit/cb9ec31bc4bd5ac5ae4836e97680e668bad19a1e#diff-d2ab9925cad7eac58e0ff4cc0d251a937ecf49e4b6bf57f8b95aab76648a9d34) ended up removing compatibility with `illuminate/contracts` below `^7.0`.
Which means, I believe, **v2.1** dropped support for anything below Laravel 7, not 6 as state in the [tag comment](https://github.com/spatie/laravel-honeypot/releases/tag/2.1.0).
Was that intentional/necessary @freekmurze?

**Reply by blakeyman** [reply]
I suggest upgrading to Laravel 8 (I did - then it installed fine)
https://laravel.com/docs/7.x/upgrade
https://laravel.com/docs/8.x/upgrade#upgrade-8.0

**Comment by StuartCreed** [comment]
I have the same issue -> using Laravel 7.x

**Reply by blakeyman** [reply]
I suggest upgrading to Laravel 8 (I did - then it installed fine)
https://laravel.com/docs/8.x/upgrade#upgrade-8.0
